Addition of field in form schema and correcting the test cases accordingly

denduluriramasaimohanakhil opened the pull request:
dev:
  lint:0
  coverage:85%
filterChanges:
  lint:0
  coverage:85%

Added pushToElastic field in schema

Mohanakhil committed:
dev:
  lint:0
  coverage:85%
filterChanges:
  lint:0
  coverage:85%

Added pushToElastic field in schema

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java:
```diff
@@ -0,0 +1,35 @@
+    @Execution
+    public void setElasticPushField()
+    {
+        Query query = new Query();
```
this should be protected and only happen if the elastic enabled environment is set

denduluriramasaimohanakhil replied:
Ok modifying it accordingly now

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/changelog/AddGroupTest.java:
```diff
@@ -0,0 +1,47 @@
+import static org.mockito.Mockito.times;
+
+@ExtendWith(MockitoExtension.class)
+class AddGroupTest
```
why are these tests being added?

denduluriramasaimohanakhil replied:
To improve the code coverage to above 85% I have added above test cases extra

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java:
```diff
@@ -0,0 +1,35 @@
+@ChangeUnit(id="push-to-elastic",order = "5")
+@Slf4j
+@RequiredArgsConstructor
+public class Elastic
```
Give a proper name

denduluriramasaimohanakhil replied:
Yes changed the class name now

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:
```diff
@@ -184,7 +185,7 @@ void saveFormDataWithoutCollectionTest() throws IOException
                 TEST_VERSION,testFormData,testFormMetaData);
         doReturn(userList).when(mockUserDetails).getUserDetails();
         FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
```
replace this constructor

denduluriramasaimohanakhil replied:
Yes Replacing with No argument constructor

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledExceptionTest.java:
```diff
@@ -124,7 +125,7 @@ void saveFormDataWithoutCollectionInvalidInputExceptionTest() throws IOException
         doReturn(userList).when(mockUserDetails).getUserDetails();
         FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
-                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON));
+                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON), Status.DISABLED);
```
this need not be provided to the constructor

denduluriramasaimohanakhil replied:
Correcting as above

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:
```diff
@@ -250,7 +251,7 @@ void saveFormDataWithUniqueKeyCheckCollectionNotExist() throws IOException
         doReturn(userList).when(mockUserDetails).getUserDetails();
         FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
-                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON));
+                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON),Status.DISABLED);
```
refactor to not user constructor

denduluriramasaimohanakhil replied:
Yes doing the same as above

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:
```diff
@@ -420,6 +421,9 @@ void saveFormDataEmptyDocumentIdTest() throws IOException
         document.append("version",1);
+        FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
+                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID,String.valueOf(TEST_UPDATED_ON),Status.DISABLED);
+        Mockito.when(mockFormService.getRuntimeFormById(anyString())).thenReturn(formResponseSchemaTest);
```
refactor to replace the constructor

denduluriramasaimohanakhil replied:
Yes doing the same as above

Mohanakhil committed:
dev:
  lint:0
  coverage:85%
pushDataToElastic:
  lint:0
  coverage:85%

Added elastic service as aspect and corrected unittestcases accordingly

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java:
```diff
@@ -0,0 +1,35 @@
+    public void setElasticPushField()
+    {
+        Query query = new Query();
+        Update update = new Update().set("elasticPush", Status.ENABLED);
```
this value needs to be set as per the environment variable value

denduluriramasaimohanakhil replied:
ok vaibhav modifying it as per env value

Mohanakhil committed:
dev:
  lint:0
  coverage:85%
pushDataToElastic:
  lint:0
  coverage:85%

Resolved comments

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataElasticAspectHandler.java:
```diff
@@ -0,0 +1,35 @@
+    private final ObjectMapper objectMapper;
+    private final FormDataElasticService formDataElasticService;
+
+    @Before("execution(* com.techsophy.tsf.runtime.form.controller.FormDataController..*(..))")
```
just explain if we need before or after, what would match the current behavior

denduluriramasaimohanakhil replied:
I will explain vaibhav

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/entity/Auditable.java:
```diff
@@ -23,4 +14,6 @@
     private String updatedById;
     @LastModifiedDate
     private String updatedOn;
+    @Version
```
dont use audit version, lets set created attributes manually

denduluriramasaimohanakhil replied:
Ok sir will go manually only

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java:
```diff
@@ -7,24 +7,32 @@
 @RequiredArgsConstructor
 public class ElasticFieldMigration
 {
+    @Value(ELASTIC_ENABLE)
```
pls dont add the annotations via constants

denduluriramasaimohanakhil replied:
Yes replacing with string only

Mohanakhil committed:
dev:
  lint:0
  coverage:85%
pushDataToElastic:
  lint:0
  coverage:85%

Resolved comments

Mohanakhil committed:
dev:
  lint:0
  coverage:85%
pushDataToElastic:
  lint:0
  coverage:85%

Resolved comments

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java:
```diff
@@ -0,0 +1,44 @@
+        FormDataElasticServiceImpl mockFormDataElasticServiceImpl=
+                new FormDataElasticServiceImpl(mockWebClientWrapper,mockTokenUtils,gatewayURL,elasticSource);
+        FormDataDefinition formDataDefinition=new FormDataDefinition();
+        Assertions.assertDoesNotThrow(()->mockFormDataElasticServiceImpl.saveOrUpdateToElastic(formDataDefinition));
```
this is not an appropriate test case with mocked data. You should verify that the downstream is called with expected inputs at least.

denduluriramasaimohanakhil replied:
corrected this testcase

Mohanakhil committed:
dev:
  lint:0
  coverage:85%
pushDataToElastic:
  lint:0
  coverage:85%

Resolved comments
